databricks/koalas/tests/test_indexes.py

```diff
@@ -524,3 +524,66 @@ def test_argmax(self):
                 TypeError,
                 "reduction operation 'argmax' not allowed for this dtype"):
             kidx.argmax()
+
+    def test_monotonic(self):
+        # test monotonic_increasing & monotonic_decreasing for MultiIndex
+        datas = []
+
+        # increasing / decreasing ordered each index level with string
+        datas.append([('w', 'a'), ('x', 'b'), ('y', 'c'), ('z', 'd')])
+        datas.append([('w', 'd'), ('x', 'c'), ('y', 'b'), ('z', 'a')])
+        datas.append([('z', 'a'), ('y', 'b'), ('x', 'c'), ('w', 'd')])
+        datas.append([('z', 'd'), ('y', 'c'), ('x', 'b'), ('w', 'a')])
+        # mixed order each index level with string
+        datas.append([('z', 'a'), ('x', 'b'), ('y', 'c'), ('w', 'd')])
+        datas.append([('z', 'a'), ('y', 'c'), ('x', 'b'), ('w', 'd')])
+
+        # increasing / decreasing ordered each index level with integer
+        datas.append([(1, 100), (2, 200), (3, 300), (4, 400), (5, 500)])
+        datas.append([(1, 500), (2, 400), (3, 300), (4, 200), (5, 100)])
+        datas.append([(5, 100), (4, 200), (3, 300), (2, 400), (1, 500)])
+        datas.append([(5, 500), (4, 400), (3, 300), (2, 200), (1, 100)])
+        # mixed order each index level with integer
+        datas.append([(1, 500), (3, 400), (2, 300), (4, 200), (5, 100)])
+        datas.append([(1, 100), (2, 300), (3, 200), (4, 400), (5, 500)])
+
+        # negative tests
+        datas.append([(-1, -100), (-2, -200), (-3, -300), (-4, -400), (-5, -500)])
+        datas.append([(-1, -500), (-2, -400), (-3, -300), (-4, -200), (-5, -100)])
+        datas.append([(-5, -100), (-4, -200), (-3, -300), (-2, -400), (-1, -500)])
+        datas.append([(-5, -500), (-4, -400), (-3, -300), (-2, -200), (-1, -100)])
+
+        # integer / negative mixed tests
+        datas.append([('a', -500), ('b', -400), ('c', -300), ('d', -200), ('e', -100)])
+        datas.append([('e', -500), ('d', -400), ('c', -300), ('b', -200), ('a', -100)])
+        datas.append([(-5, 'a'), (-4, 'b'), (-3, 'c'), (-2, 'd'), (-1, 'e')])
+        datas.append([(-5, 'e'), (-4, 'd'), (-3, 'c'), (-2, 'b'), (-1, 'a')])
+        datas.append([(-5, 'e'), (-3, 'd'), (-2, 'c'), (-4, 'b'), (-1, 'a')])
+        datas.append([(-5, 'e'), (-4, 'c'), (-3, 'b'), (-2, 'd'), (-1, 'a')])
+
+        # None type tests
+        # TODO: pandas returns `True` commented cases below
+        # since they ignore the `None` positioned last.
+        datas.append([(None, 100), (2, 200), (3, 300), (4, 400), (5, 500)])
+        datas.append([(1, 100), (2, 200), (None, 300), (4, 400), (5, 500)])
+        # datas.append([(1, 100), (2, 200), (3, 300), (4, 400), (None, 500)])
```

Comment:
we have some tricky problem here.

if `MultiIndex` has `None` in a row at the last, we should ignore them.

for example.

```python
>>> pidx
MultiIndex([(1.0, 100),
            (2.0, 200),
            (3.0, 300),
            (4.0, 400),
            (nan, 500)],
           )
>>> pidx.is_monotonic
True

>>> pidx
MultiIndex([(1.0, 100),
            (2.0, 200),
            (3.0, 300),
            (nan, 400),
            (nan, 500)],
           )
>>> pidx.is_monotonic
True

>>> pidx
MultiIndex([(1.0, 100),
            (2.0, 200),
            (nan, 300),
            (nan, 400),
            (nan, 500)],
           )
>>> pidx.is_monotonic
True
```

but to this, i think we have to traverse all data and it might possibly quite expensive.

cc @ueshin @HyukjinKwon could you share the idea to solve this if when you available??